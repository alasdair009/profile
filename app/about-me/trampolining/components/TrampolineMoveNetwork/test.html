<html lang="en">
  <head>
    <style>
      .videos {
        display: flex;
        height: 100vh;
        width: 100vw;
      }
      .videos div {
        align-items: center;
        display: flex;
        flex-direction: column;
        justify-content: center;
        width: 50%;
      }
      iframe {
        aspect-ratio: 16/9;
        width: 90%;
      }
    </style>
  </head>
  <body>
    <div class="videos">
      <div>
        <iframe
          id="standard"
          src=""
          data-src="https://www.youtube.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Standard Embed"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goStandard">Standard</button>
      </div>
      <div>
        <iframe
          id="privacy"
          src=""
          data-src="https://www.youtube-nocookie.com/embed/NpEaa2P7qZI?si=uJjK4C2TLWivF5BX"
          title="Enhanced privacy"
          allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share"
          referrerpolicy="strict-origin-when-cross-origin"
          allowfullscreen
        ></iframe>
        <button id="goPrivacy">Privacy</button>
      </div>
    </div>
    <script>
      const goStandard = document.getElementById("goStandard");
      const standardIFrame = document.getElementById("standard");
      const goPrivacy = document.getElementById("goPrivacy");
      const privacyIFrame = document.getElementById("privacy");
      goStandard.addEventListener(
        "click",
        function () {
          standardIFrame.setAttribute("src", standardIFrame.getAttribute("data-src"));
        },
        false
      );
      goPrivacy.addEventListener(
        "click",
        function () {
          privacyIFrame.setAttribute("src", privacyIFrame.getAttribute("data-src"));
        },
        false
      );
    </script>
  </body>
</html>
