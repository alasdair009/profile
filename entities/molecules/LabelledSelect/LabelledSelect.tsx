import { LabelText, Root } from "./styles";
import { HTMLAttributes } from "react";
import { ErrorText } from "@/entities";
import { BaseSelect } from "@/entities/atoms/BaseSelect";

type LabelledSelectProps = {
  /**
   * Text for the label.
   */
  label: string;
  /**
   * Name of the field.
   */
  name: string;
  /**
   * Field requires a value to be valid.
   */
  required?: boolean;
  /**
   * If multiple entries can be chosen.
   */
  multiple?: boolean;
  /**
   * Field displays as an invalid input.
   */
  isInvalid?: boolean;
  /**
   * Error text the field should display.
   */
  errorText?: string;
} & HTMLAttributes<HTMLSelectElement>;

/**
 * Labelled text input field for forms.
 */
export function LabelledSelect({
  isInvalid,
  name,
  label,
  errorText = `Please enter a valid ${label}`,
  children,
  ...rest
}: LabelledSelectProps) {
  return (
    <Root data-testid={LabelledSelect.name}>
      <LabelText>{label}</LabelText>
      <BaseSelect isInvalid={isInvalid} {...rest}>
        {children}
      </BaseSelect>
      <ErrorText shown={isInvalid || false}>{errorText}</ErrorText>
    </Root>
  );
}
