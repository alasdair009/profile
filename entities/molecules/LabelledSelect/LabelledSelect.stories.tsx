import { LabelledSelect } from "./LabelledSelect";
import { Meta, StoryObj } from "@storybook/react";
import { expect, userEvent, within } from "@storybook/test";

const meta: Meta<typeof LabelledSelect> = {
  component: LabelledSelect,
  args: {
    name: "selectentry",
    required: true,
    isInvalid: false,
    label: "Select entry",
    errorText: "",
    children: (
      <>
        <option value="value 1" data-testid="option">
          option 1
        </option>
        <option value="value 2" data-testid="option">
          option 2
        </option>
        <option value="value 3" data-testid="option">
          option 3
        </option>
      </>
    ),
  },
};
export default meta;

export const Default: StoryObj<typeof LabelledSelect> = {
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledSelectElement = canvas.getAllByTestId(LabelledSelect.name)[0];
    const baseSelectElement = canvas.getByLabelText(args.label);
    const labelElement = canvas.getByText(args.label);

    await expect(labelledSelectElement).toBeInTheDocument();
    await expect(baseSelectElement).toBeInTheDocument();
    await expect(labelElement).toBeInTheDocument();

    await userEvent.click(baseSelectElement);
  },
};

export const Multiple: StoryObj<typeof LabelledSelect> = {
  args: {
    multiple: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledSelectElement = canvas.getAllByTestId(LabelledSelect.name)[0];
    const baseSelectElement = canvas.getByLabelText(args.label);
    const labelElement = canvas.getByText(args.label);

    await expect(labelledSelectElement).toBeInTheDocument();
    await expect(baseSelectElement).toBeInTheDocument();
    await expect(labelElement).toBeInTheDocument();

    await userEvent.click(baseSelectElement);
  },
};

export const Invalid: StoryObj<typeof LabelledSelect> = {
  args: {
    isInvalid: true,
  },
  play: async ({ args, canvasElement }) => {
    const canvas = within(canvasElement);
    const labelledSelectElement = canvas.getAllByTestId(LabelledSelect.name)[0];
    const baseSelectElement = canvas.getByLabelText(args.label);
    const labelElement = canvas.getByText(args.label);

    await expect(labelledSelectElement).toBeInTheDocument();
    await expect(baseSelectElement).toBeInTheDocument();
    await expect(labelElement).toBeInTheDocument();

    await userEvent.click(baseSelectElement);
  },
};
