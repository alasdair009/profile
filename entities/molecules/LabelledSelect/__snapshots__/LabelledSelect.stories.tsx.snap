// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`molecules/LabelledSelect Default play-test 1`] = `
<label data-testid="LabelledSelect"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Select entry
  </span>
  <select data-testid="BaseSelect"
          required
          class="sc-iKTcqh dLFXfh"
  >
    <option value="value 1"
            data-testid="option"
    >
      option 1
    </option>
    <option value="value 2"
            data-testid="option"
    >
      option 2
    </option>
    <option value="value 3"
            data-testid="option"
    >
      option 3
    </option>
  </select>
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;

exports[`molecules/LabelledSelect Invalid play-test 1`] = `
<label data-testid="LabelledSelect"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Select entry
  </span>
  <select data-testid="BaseSelect"
          required
          class="sc-iKTcqh fpeCBF"
  >
    <option value="value 1"
            data-testid="option"
    >
      option 1
    </option>
    <option value="value 2"
            data-testid="option"
    >
      option 2
    </option>
    <option value="value 3"
            data-testid="option"
    >
      option 3
    </option>
  </select>
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH cfxBNa"
  >
  </span>
</label>
`;

exports[`molecules/LabelledSelect Multiple play-test 1`] = `
<label data-testid="LabelledSelect"
       class="sc-hDcvty eRBMpz"
>
  <span data-testid="BaseLabel"
        class="sc-fQpRED fajILV"
  >
    Select entry
  </span>
  <select multiple
          data-testid="BaseSelect"
          required
          class="sc-iKTcqh kLQcNu"
  >
    <option value="value 1"
            data-testid="option"
    >
      option 1
    </option>
    <option value="value 2"
            data-testid="option"
    >
      option 2
    </option>
    <option value="value 3"
            data-testid="option"
    >
      option 3
    </option>
  </select>
  <span color="#df1c41"
        font-size="small"
        data-testid="ErrorText"
        class="sc-dwYcXH kKOlQN"
  >
  </span>
</label>
`;
